kythe/typescript/indexer.ts

```diff
@@ -351,6 +359,9 @@ class Visitor {
           const decl = node as ts.NamedDeclaration;
           if (decl.name && decl.name.kind === ts.SyntaxKind.Identifier) {
             let part = decl.name.text;
+            if (this.isStaticMember(startNode, decl)) {
+              part += '#type';
```

Comment:
I think this might be backwards.

Given a class A, the value `A` has all the statics as properties.  Confusingly the type is `typeof A`, not `A`.  Meanwhile, *instances* of `A` (which are of type `A`) are the things that carry the members on them.

Another way to say this is like:

```
const obj = { z: 3 };
class C1 {
  z: number;
}
class C2 {
  static z: number;
}
interface HasZ {
  z: number;
}
let hasZ: HasZ = obj;  // valid
hasZ = C1;  // wrong
hasZ = C2;  // valid
hasZ = new C1;  // valid
```

And for the `HasZ` case we do name the field like `HasZ.z` or so on, so I think the class should match.


One other thing to check here:

```
class X {
  static y = class Y {
    z: number;
  };
  static a = {
    b: 3;
  };
}
```

I'm not sure what name we get for `z` and `b`.  (It could be that we do something entirely wrong already and so don't worry about it in that case, I just wanna be sure we don't do something catastrophically wrong.)

Reply:
With the new changes, for z, we get a signature of `X#type.y.Y.z`. For b, it's `X.a.anon0.b` because of the anonymous block.

Reply:
I think separately from this PR, we should reconsider how both these signatures are created.

> class X {
>   static y = class Y {
>     z: number;
>   };
>   static a = {
>     b: 3;
>   };
> }

Ideally, `z` would have the signature `X.y.z` and `b` `X.a.b`. Right now the problem is we're generating extra scopes where maybe we don't want them (at assignments?)

Reply:
A code like

```typescript
const a = class A {
  prop = 0;
};
```

Creates a declaration file

```typescript
declare const a: {
    new (): {
        prop: number;
    };
};
```

I'm not sure how to resolve this.

Should we move this discussion to a new issue?

Reply:
Moved to a separate issue. Resolving this conversation.